frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditCodeList/OptionListEditor.test.tsx

```diff
@@ -0,0 +1,171 @@
+    });
+
+    await openModal(user);
+
+    expect(screen.getByRole('dialog')).toBeInTheDocument();
+  });
+
+  it('should close Dialog', async () => {
+    const user = userEvent.setup();
+    renderOptionListEditor();
+    await waitForElementToBeRemoved(() => {
+      return screen.queryByText(textMock('ux_editor.modal_properties_code_list_spinner_title'));
+    });
+
+    await openModal(user);
+
+    await user.click(screen.getByRole('button', { name: 'close modal' })); // Todo: Replace "close modal" with defaultDialogProps.closeButtonTitle when https://github.com/digdir/designsystemet/issues/2195 is fixed
+
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+  });
+
+  it('should call handleClose when closing Dialog', async () => {
+    const user = userEvent.setup();
+    const doReloadPreview = jest.fn();
+    renderOptionListEditor({
+      previewContextProps: { doReloadPreview },
+    });
+    await waitForElementToBeRemoved(() => {
+      return screen.queryByText(textMock('ux_editor.modal_properties_code_list_spinner_title'));
+    });
+
+    await openModal(user);
+
+    await user.click(screen.getByRole('button', { name: 'close modal' })); // Todo: Replace "close modal" with defaultDialogProps.closeButtonTitle when https://github.com/digdir/designsystemet/issues/2195 is fixed
+
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+    expect(doReloadPreview).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call onChange when changing value in a textField', async () => {
+    const user = userEvent.setup();
+    const doReloadPreview = jest.fn();
+    renderOptionListEditor({
+      previewContextProps: { doReloadPreview },
+    });
+    await waitForElementToBeRemoved(() => {
+      return screen.queryByText(textMock('ux_editor.modal_properties_code_list_spinner_title'));
+    });
+
+    await openModal(user);
+
+    const textBox = screen.getByRole('textbox', {
+      name: textMock('ux_editor.modal_properties_code_list_item_description', { number: 2 }),
+    });
+    expect(textBox).toHaveTextContent('');
+    await user.type(textBox, 'test');
+
+    expect(
+      screen.getByRole('textbox', {
+        name: textMock('ux_editor.modal_properties_code_list_item_description', { number: 2 }),
+      }),
+    ).toHaveValue('test');
+  });
+});
+
+const openModal = async (user: UserEvent) => {
+  const btnOpen = screen.getByRole('button', {
+    name: textMock('ux_editor.modal_properties_code_list_open_editor'),
+  });
+  await user.click(btnOpen);
+};
+
+const renderOptionListEditor = ({
+  previewContextProps = {},
+  queries = {
+    getOptionLists: jest
+      .fn()
+      .mockImplementation(() => Promise.resolve<Record<string, Option[]>>(apiResult)),
+  },
+} = {}) => {
+  return renderWithProviders(
+    <OptionListEditor
+      component={{
+        ...mockComponent,
+      }}
+    />,
+    {
+      queries: queries,
```

Comment:
Here we can also define the getOptionLists query here:
`queries: { getOptionLists, ...queries },`